bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/image/StoredImagesCache.scala

```diff
@@ -0,0 +1,161 @@
+package edu.gemini.catalog.image
+
+import java.io.File
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.{FileVisitResult, Files, Path, SimpleFileVisitor}
+import java.time.Instant
+
+import edu.gemini.spModel.core.Angle
+import jsky.util.Preferences
+import squants.information.Information
+
+import scalaz._
+import Scalaz._
+import scalaz.concurrent.Task
+
+/**
+  * Keeps track of images in the file system and access time
+  */
+protected case class StoredImages(entries: List[(Instant, ImageInFile)]) {
+  def images: List[ImageInFile] = entries.map(_._2)
+
+  def +(i: ImageInFile): StoredImages             = copy((Instant.now, i) :: entries)
+  def +(i: Instant, e: ImageInFile): StoredImages = copy((i, e) :: entries)
+  def -(i: ImageInFile): StoredImages             = copy(entries.filterNot(_._2 === i))
+
+  /**
+    * Indicates the image was used, update the access time
+    */
+  def touch(i: ImageInFile): StoredImages = copy(entries.collect {
+    case x if x._2 === i => (Instant.now, x._2)
```

Comment:
Yes. Uppercase and back-quoted identifiers are matched by value.

```scala
case  Foo  => ... matches only the value Foo and binds nothing
case  foo  => ... matches everything and binds foo
case `foo` => ... matches only the value foo and binds nothing
```

It's a little-known feature. Note that it uses `==` which sucks.